docs/working_with_splat.md

```diff
@@ -0,0 +1,122 @@
+
+## Importing data and bss
+
+The data section that was extracted from the rom can be found in `<file_name>.data.s`.
+
+To start out with you can copy the hexadecimal numbers into your c file and format the data as an array of s32s. For example:
+
+```c
+s32 D_80AAC084_jp[] = {0x01000100, 0x00010100, 0x01000000, 0x01010001, 0x01000100};
+```
+
+If you know the proper type for this data, you can define it as that and split the hexadecimal numbers accordingly.
+
+Once you have copied over all the data, edit `boot.yaml` or `code.yaml` depending on your file. you want to change your file's `data` definition to `.data` so splat will no longer extract it. Save the file, and run `make extract -j` and `make -j`. We're checking to see if the rom matches after building. If it doesn't, it's likely that something about the data you imported is incorrect. `python3 tools/first_diff.py` and `vbindiff` can be helpful for diagnosing problems.
+
+Importing bss is similar. Declare the bss variable in your C file, in the yaml change `bss` to `.bss`, and extract and build again to make sure the rom matches. With bss you have to make sure it gets ordered correctly, and unfortunately some files can be uncooperative. Sometimes hacks are required, like using `prevent_bss_reordering.h`.
+
+## Renaming Functions and Symbols.
+
+Open `linkerscripts/jp/symbol_addrs_boot.txt` or `linkerscripts/jp/symbol_addrs_code.txt` depending on your file. Every function and symbol has a definition already, so search for the default func_XXXXXXXX or D_XXXXXXXX, and replace it with your new name.
+
+For data, sometimes it will get extracted incorrectly, splitting up data or adding padding at the end. You should manually remove unnecessary definitions. 
+
+Save the file. Like the yaml files, after editing these you need to run `make extract -j` again. Because your C file already exists, splat will not generate a new one. You will need to manually rename everything in your C and header file.
+
+## Actors and Overlays
+
+Because the n64 has a very small amount of ram, the game's code isn't entirely loaded into memory at once. Instead, certain files get loaded when they're needed, and unloaded when they aren't. These files are called overlays. Every actor is an overlay, but not every overlays is an actor.
+
+Because an overlay doesn't know what arbitrary spot in ram it will be loaded into, there's a system that remaps the virtual ram addresses to physical ram addresses. This is handled by the .reloc section of the file.
+
+Because of virtual ram, `yamls/jp/overlays.yaml` is set up a little bit different from the others. Each file has it's own definition, and the text, data, rodata, and bss are defined as subsegments. This allows for manually specifying the virtual ram address of each file with the `vram` argument.
+
+For the purpose of decompiling overlays, editing the subsegment definitions is the same as code and boot files. You should not have to change the reloc file definitions.

```

Comment:
We will eventually want to generate relocs instead of just extracting them, but I think it doesn't matter right now and it can be edited when we actually can generate relocs